doc/source/blueprint_cce_vault.rst

```diff
@@ -0,0 +1,592 @@
+               - /config-scripts/ok
+             failureThreshold: 2
+             initialDelaySeconds: 20
+             periodSeconds: 30
+             successThreshold: 1
+             timeoutSeconds: 5
+           name: zookeeper
+           ports:
+           - containerPort: 2281
+             name: client
+             protocol: TCP
+           - containerPort: 2888
+             name: server
+             protocol: TCP
+           - containerPort: 3888
+             name: election
+             protocol: TCP
+           readinessProbe:
+             exec:
+               command:
+               - sh
+               - /config-scripts/ready
+             failureThreshold: 2
+             initialDelaySeconds: 20
+             periodSeconds: 30
+             successThreshold: 1
+             timeoutSeconds: 5
+           securityContext:
+             runAsUser: 1000
+           volumeMounts:
+           - mountPath: /data
+             name: datadir
+           - mountPath: /tls
+             name: cert-data
+           - mountPath: /config-scripts
+             name: zookeeper-config
+         dnsPolicy: ClusterFirst
+
+         initContainers:
+         - args:
+           - agent
+           - -config=/etc/vault/vault-agent-config.hcl
+           - -log-level=debug
+           - -exit-after-auth=true
+           env:
+           - name: VAULT_ADDR
+             value: <VAULT_PUBLIC_ADDR>
+           image: vault:1.9.0
+           name: vault-agent
+           volumeMounts:
+           - mountPath: /etc/vault
+             name: vault-agent-config
+           - mountPath: /tls
+             name: cert-data
+           - mountPath: /var/run/secrets/tokens
+             name: k8-tokens
+         restartPolicy: Always
+         serviceAccount: zookeeper
+         serviceAccountName: zookeeper
+         terminationGracePeriodSeconds: 1800
+         volumes:
+         - configMap:
+             defaultMode: 420
+             items:
+             - key: vault-agent-config.hcl
+               path: vault-agent-config.hcl
+             name: zookeeper-config
+           name: vault-agent-config
+         - configMap:
+             defaultMode: 365
+             name: zookeeper-config
+           name: zookeeper-config
+         - emptyDir: {}
+           name: cert-data
+         - name: k8-tokens
+           projected:
+             defaultMode: 420
+             sources:
+             - serviceAccountToken:
+                 expirationSeconds: 7200
+                 path: vault-token
+
+     updateStrategy:
+       type: RollingUpdate
+     volumeClaimTemplates:
+     - apiVersion: v1
+       kind: PersistentVolumeClaim
+       metadata:
+         name: datadir
+       spec:
+         accessModes:
+         - ReadWriteOnce
+         resources:
+           requests:
+             storage: 5Gi
+         storageClassName: csi-disk
+         volumeMode: Filesystem
+
+.. code-block:: bash
+
+   $ kubectl apply -f zookeeper-ss.yaml
+
```

Comment:
I have the feeling that the code snippets are disbalanced long compared to the text explanations. How did you come up with what is in the scripts? Did you just use some defaults and adapted them? Did you follow a reference manual step by step to fill in all those attributes?

If there's a reason for this being so long, that's fine, but then please explain it.